reflection/Reflection/Reflection/classes/ConfigurationManagerConfigurationProvider.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Configuration;
+
+namespace Reflection.classes
+{
+    public class ConfigurationManagerConfigurationProvider : IConfigurationProvider
+    {
+        public void SaveSetting(string settingName, object value)
+        {
+            try
+            {
+                Configuration config = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+                KeyValueConfigurationElement setting = config.AppSettings.Settings[settingName];
+                if (setting != null)
+                {
+                    setting.Value = value.ToString();
+                    config.Save(ConfigurationSaveMode.Modified);
+                    ConfigurationManager.RefreshSection("appSettings");
+                }
+                else
+                {
+                    Console.WriteLine($"Setting '{settingName}' not found in configuration file.");
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Error saving setting: {e.Message}");
+            }
+        }
+
+        public object GetSetting(string settingName, Type type)
+        {
+            try
+            {
+                string value = ConfigurationManager.AppSettings[settingName];
+                return Convert.ChangeType(value, type);
```

Comment:
Convert.ChangeType() method doesn't provide a direct conversion from string to TimeSpan, so it will always fail on such convertion. Add condition for TimeSpan Type and use TimeSpan.Parse() to convert from string instead of Convert.ChangeType().
Example below:
```
switch (type)
{
    case var _ when type == typeof(TimeSpan):
        return TimeSpan.Parse(value);
    default:
         return Convert.ChangeType(value, type);
}
```
(or use simple if statement)